expression/builtin_time_vec.go

```diff
@@ -787,11 +787,49 @@ func (b *builtinUnixTimestampDecSig) vecEvalDecimal(input *chunk.Chunk, result *
 }
 
 func (b *builtinPeriodAddSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinPeriodAddSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	periods := buf.Int64s()
+	ns := buf1.Int64s()
+	result.ResizeInt64(n, false)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) || buf1.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+
+		if !validPeriod(periods[i]) {
+			if err := errIncorrectArgs.GenWithStackByArgs("period_add"); err != nil {
+				return err
+			}
+			i64s[i] = 0
+			continue
```

Comment:
The `err` will never be `nil` here, so it's ok to return `errIncorrectArgs.GenWithStackByArgs("period_add")` directly.